File: src/Point.java
// Java:
public class Point {
  private double x, y;
 
  public Point(final double X, final double Y) {
    x = X;
    y = Y;
  }
 
  public double x() {
    return x;
  }
 
  public double y() {
    return y;
  }
 
  public Point(final double X,
    final double Y,
        final boolean ADD2GRID) {
    this(X, Y);
 
    if (ADD2GRID)
			grid.add(this);
  }
 
  public Point() {
    this(0.0, 0.0);
  }
 
  double distanceToPoint(final Point OTHER) {
    return distanceBetweenPoints(x, y,
      OTHER.x, OTHER.y);
  }
 
  private static Grid grid = new Grid();
 
  static double distanceBetweenPoints(
      final double X1, final double Y1,
      final double X2, final double Y2
  ) {
    double xDist = X1 - X2;
    double yDist = Y1 - Y2;
    return Math.sqrt(xDist*xDist + yDist*yDist);
  }}

File: src/Point.scala
// Scala
class Point(
  // adding `val` here automatically creates
  // public accessor methods named `x` and `y`
  val x: Double = 0.0, val y: Double = 0.0,
  addToGrid: Boolean = false) {

  // import functions/vars from companion object
  //if (addToGrid)  Point.grid.add(this)
  def distanceToPoint(other: Point) =
    Point.distanceBetweenPoints(x, y, other.x, other.y)
}

object Point {
  // private/protected members shared between
  // class and companion object
  //private val grid = new Grid()
  def distanceBetweenPoints(
    x1: Double, y1: Double,
    x2: Double, y2: Double) = {
    val xDist = x1 - x2
    val yDist = y1 - y2
    math.sqrt(xDist * xDist + yDist * yDist)
  }
}

class Ellips(middelpunt: Point, kort: Int, lang: Int)
class Cirkel(middelpunt: Point, straal: Int) extends Ellips(middelpunt, straal, straal) {
  def omtrek = 2 * straal * Cirkel.pi
  private val a = 0
}
object Cirkel {
  val b = Cirkel
  val pi = 3.14
}
